packages/app/src/components/catalog/EntityPage/ContextMenuAwareEntityLayout.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { ReactNode, useMemo, useState } from 'react';
+import { EntityLayout } from '@backstage/plugin-catalog';
+import getMountPointData from '../../../utils/dynamicUI/getMountPointData';
+import { MenuIcon } from '../../Root/Root';
+
+export const ContextMenuAwareEntityLayout = (props: {
+  children?: ReactNode;
+}) => {
+  const contextMenuElements =
+    getMountPointData<React.ComponentType<React.PropsWithChildren<any>>>(
+      `entity.context.menu`,
+    );
+
+  const [openStates, openStatesSet] = useState(
+    new Array<boolean>(contextMenuElements.length).fill(false),
+  );
+
+  const extraMenuItems = useMemo(
```

Comment:
I wonder if this useMemo is really needed.  It's worth removing this and seeing if the behavior changes at all, I suspect it's not adding anything.